client/src/components/Invitations/InvitationDialog.js

```diff
@@ -0,0 +1,281 @@
+        margin: theme.spacing(1),
+      },
+    },
+    dialogContentTextRoot: {
+      fontWeight: 'bold',
+      color: theme.palette.text.primary,
+      marginBottom: 0,
+    },
+    dialogActionsRoot: {
+      justifyContent: 'center',
+      margin: theme.spacing(3),
+    },
+    btn: {
+      fontWeight: '600'
+    },
+    btnMixedCase: {
+      textTransform: 'unset !important',
+      fontWeight: '600',
+      fontSize: theme.spacing(2),
+    },
+    btnMixedCaseCenter: {
+      textTransform: 'unset !important',
+      fontWeight: '600',
+      fontSize: theme.spacing(2),
+      alignItem: 'center',
+    },
+    closeButton: {
+      position: 'absolute',
+      right: theme.spacing(1),
+      top: theme.spacing(1),
+      color: theme.palette.action.disabled,
+    },
+    btnOverlay: {
+      zIndex: '2000',
+      position: 'absolute',
+      right: theme.spacing(4),
+      top: theme.spacing(32),
+    }
+  }));
+
+const dialogTitleStyles = (theme) => ({
+  root: {
+    margin: 0,
+    padding: theme.spacing(2),
+  },
+  title: {
+    color: theme.palette.text.primary,
+    align: 'center',
+    margin: theme.spacing(3),
+    fontWeight: 'bold'
+  }
+});
+
+const DialogTitle = withStyles(dialogTitleStyles)((props) => {
+  const { children, classes, ...other } = props;
+  return (
+    <MuiDialogTitle disableTypography className={classes.root} {...other}>
+      <div>
+        <Typography variant="h5" className={classes.title}
+          align="center">
+            {children}
+        </Typography>
+      </div>
+    </MuiDialogTitle>
+  );
+});
+
+export default function InvitationDialog() {
+  const [open, setOpen] = useState(false);
+  const [email, setEmail] = useState('');
+  const [emailErrorMessage, setEmailErrorMessage] = useState('');
+  const [submitError, setSubmitError] = useState('');
+  const [referralUrl, setReferralUrl] = useState('http://www.messenger.com/join/123456');
+  const classes = useStyles();
+
+  const handleChange = (event) => {
+    const {value} = event.target
+    setEmail(value.trim());
+  };
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = (ev) => {
+    setOpen(false);
+  }
+
+  //handle submit form
+  const handleSave = (ev) => {
+    ev.preventDefault();
+    //TODO
+    let jwtToken = '';
+
+    //handle 3 cases
+    //email is empty
+    //email is populated
+      //email is invalid (any email)
+      //email is valid (all)
+
+    if (!email.length) {
+      const emptyEmailError = 'Please enter an email.';
+      setEmailErrorMessage(emptyEmailError);
+    } else {
+      let emailAr = getEmailAr(email);
+      let emailsAreValid = true;
+
+      if (emailAr.length === 1) {
+        emailsAreValid = isEmailValid(email);
+      } else {
+        for (let i = 0; i < emailAr.length; i++) {
+          if (!isEmailValid(emailAr[i])) {
+            emailsAreValid = false;
+            break;
+          }
+        }  
+      }
+
+      if (!emailsAreValid) {
```

Comment:
Suggestion to rewrite the logic to reduce repeated code:

```

const emailsAreValid = (emailArrays) => {
  emailArrays.forEach( (email) => { 
    if (!isEmailValid(email) return false;
  })

  return true;
}

if (!email.length) {
} else if (!emailsAreValid(emailArray) {
} else {
  // send to backend, by this point it should have passed all validation test
}
```

Reply:
so even if email Array is length 1, going through that helper function will work

Reply:
done. that makes sense.

Reply:
revised code looks good!